spec.bs

```diff
@@ -1101,6 +1102,60 @@ For now, see [https://github.com/wicg/observable#operators](https://github.com/w
     1. <span class=XXX>TODO: Spec this and use |predicate| and |options|.</span>
 </div>
 
+<div algorithm>
+  The <dfn for=Observable method><code>first(|options|)</code></dfn> method steps are:
+
+    1. Let |p| [=a new promise=].
+
+    1. Let |controller| be a [=new=] {{AbortController}}.
+
+    1. Let |internal options| be a new {{SubscribeOptions}} whose {{SubscribeOptions/signal}} is the
+       result of [=creating a dependent abort signal=] from the list «|controller|'s
+       [=AbortController/signal=], |options|'s {{SubscribeOptions/signal}} if non-null», using
+       {{AbortSignal}}, and the [=current realm=].
+
+    1. If |internal options|'s {{SubscribeOptions/signal}} is [=AbortSignal/aborted=], then:
+
+       1. [=Reject=] |p| with |internal options|'s {{SubscribeOptions/signal}}'s
+          [=AbortSignal/abort reason=].
+
+       1. Return |p|.
+
+    1. [=AbortSignal/add|Add the following abort algorithm=] to |internal options|'s
+       {{SubscribeOptions/signal}}:
+
+       1. [=Reject=] |p| with |internal options|'s {{SubscribeOptions/signal}}'s [=AbortSignal/abort
+          reason=].
+
+    1. Let |internal observer| be a new [=internal observer=], initialized as follows:
+
+       : [=internal observer/next steps=]
+       :: 1. [=Resolve=] |p| with the passed in <var ignore>value</var>.
+          
+          1. [=AbortController/Signal abort=] |controller|.
+
+       : [=internal observer/error steps=]
+       :: [=Reject=] |p| with the passed in <var ignore>error</var>.
+
+       : [=internal observer/complete steps=]
+       :: [=Reject=] |p| with a [=new=] {{RangeError}}.
```

Comment:
Question: should the message be outlined in the spec? "No values in sequence", "No values observed", or some such thing?

Reply:
I don't _think_ so, given https://html.spec.whatwg.org/C#dom-context-2d-roundrect for example.